docs/api/search.md

```diff
@@ -122,3 +127,37 @@ SELECT * FROM vectorize.search(
  {"product_id": 4, "product_name": "Bluetooth Speaker", "similarity_score": 0.8250355616233103}
 (3 rows)
 ```
+
+## Filtering Search Results
+
+The `where` parameter allows to apply SQL-based filtering after performing the vector similarity search. This feature is useful when you want to narrow down the search results based on certain conditions such as `product category` or `price`.
+
+### Example
+
+```sql
+SELECT * FROM vectorize.search(
+    job_name        => 'product_search',
+    query           => 'mobile electronic devices',
+    return_columns  => ARRAY['product_id', 'product_name'],
+    num_results     => 3,
+    where       => 'product_category = ''electronics'' AND price > 100'
+);
+```
+
+In the above example, the results are filtered where the `product_category` is `electronics` and the `price` is greater than 100.
+
+## Optimizing Searches with Partial Indices
+
+For improving performance when using filters, you can create partial indices. This will speed up the execution of queries with frequent conditions in the `where` parameter.
```

Comment:
```suggestion
For improving performance when using filters, you can create partial indices. This will speed up the execution of queries with frequent conditions in the `where_sql` parameter.
```